File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.java
class Solution {
    private boolean isVowel(char c){
        c = Character.toLowerCase(c);
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }
    public String reverseVowels(String s) {
                char[] s_array = s.toCharArray();
        int start = 0;
        int end = s.length() - 1;

        while (start < end){

            if (!isVowel(s_array[start])){
                start++;
                continue;
            }

            if (!isVowel(s_array[end])){
                end--;
                continue;
            }

            char temp = s_array[start];
            s_array[start] = s_array[end];
            s_array[end] = temp;

            start++;
            end--;
        }

        return new String(s_array);
    }
}

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.go
func reverseVowels(s string) string {
  charArray := []byte(s)
	left, right := 0, len(charArray)-1
	for left < right {

		if isVowel(charArray[left]) && isVowel(charArray[right]) {
			temp := charArray[left]
			charArray[left] = charArray[right]
			charArray[right] = temp
			left++
			right--
		}
		if isVowel(charArray[left]) && !isVowel(charArray[right]) {
			right--
		}
		if !isVowel(charArray[left]) && isVowel(charArray[right]) {
			left++
		}
		if !isVowel(charArray[left]) && !isVowel(charArray[right]) {
			left++
			right--
		}

	}
	return string(charArray)  
}
func isVowel(char byte) bool {
	char = byte(unicode.ToLower(rune(char)))
	return char == 'a' || char == 'e' || char == 'i' ||
		char == 'o' || char == 'u'

}